include/micm/util/matrix.hpp

```diff
@@ -24,10 +24,21 @@ namespace micm
     t.NumberOfGroups();
   };
 
+  template<class>
+  class Matrix;
+
+  using StandardDenseMatrix = Matrix<double>;
+
   /// @brief A 2D array class with contiguous memory
-  template<class T>
+  template<class T = double>
   class Matrix
   {
+    public:
+    // Diagonal markowitz reordering requires an int argument, make sure one is always accessible
+    using IntMatrix = Matrix<int>;
```

Comment:
What would be the use of `Matrix<int>`? The comment suggests it is because of `Diagonal markowitz reordering`. Would it be specific for backward euler? I am curious.

Reply:
When the solvers are made, we can choose to reorder the species matrix or not. I don't know the details but it apparently minimizes the number of zeros or something, so we should have better cache coherency or something. Any solver can use this, but the function needs a Matrix<int> to do things, and mostly we make a Matrix<double>. Since we aren't passing `template<template<class> Matrix` around anymore, I needed to do something like this so that we always have the int version available.